src/pyHepGrid/src/Backend.py

```diff
@@ -37,9 +37,81 @@ class Backend(_mode):
+            'W': self._get_data_warmup,
+            'S': self._get_data_warmup
+        }
+        self.assume_yes = False
+        self.act_only_on_done = act_only_on_done
+        self.stats_one_line = False
 
+    def __str__(self):
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def update_stdout(self):
+        """ retrieves stdout of all running jobs and store the current state
+        into its correspondent folder
+        """
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def renew_proxy(self, jobids):
+        """ renew proxy for a given job (optional) """
+        pass
+
+    def kill_job(self, jobids, jobinfo):
+        """ kills given job """
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def clean_job(self, jobids, jobinfo):
+        """ remove the sandbox of a given job (including its stdout!) """
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def cat_job(self, jobids, jobinfo, print_stderr=None, store=False):
+        """ print standard output of a given job"""
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def cat_log_job(self, jobids, jobinfo):
+        """Sometimes the std output doesn't get updated
+        but we can choose to access the logs themselves"""
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def bring_current_warmup(self, db_id):
+        """ Sometimes we want to retrieve the warmup before the job finishes """
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def status_job(self, jobids, verbose=False):
+        """ print the current status of a given job """
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def stats_job(self, dbid, do_print=True):
+        """ Given a list of jobs, returns the number of jobs which
+        are in each possible state (done/waiting/running/etc)
+        """
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
+    def _format_args(self, *args, **kwargs):
+        raise NotImplementedError("{0} not implemented for {1}".format(
+            sys._getframe().f_code.co_name, self.__class__.__bases__[0]))
+
```

Comment:
With these changes, do all methods defined for subclasses of Backend now have corresponding parent methods here?

Reply:
Should be, at least for the `backendManagement.py` I went through the list. I will double check for `runArcjob.py` etc.

Also why do we even have the `RunArc(Backend)` shouldn't that be a strict subclass of `Arc(Backend)`?

Reply:
I'm not sure why they should be separate at all, to be honest.

Reply:
I think it made sense in the past, the idea being that the importance was on how you interact with the storage rather than with the computing element (both run classes were doing the same thing and at the end running the correct script).

Also the backendmanagement functions were completely separated of the running ones (you would never use both of them in one single call to pyhepgrid!)

That's the history though. I guess at this point it makes more sense to have `backend -> Arc(backend) -> management(Arc) and run(Arc)` or even just `Arc(backend)`.